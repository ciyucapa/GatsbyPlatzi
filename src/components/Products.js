import React from 'react';
import { Link } from 'gatsby';
import PropTypes from "prop-types";

import {StyledProducts} from '../styles/components'

const Products = (props) => (
    <StyledProducts>
        <h2>Productos</h2>
        <section>
            {props.edges.map(({node}) => 
                <article key={node.id} >
                    <img src={node.image} alt={node.product.name} />
                    <p>{node.product.name}</p>
                    <small>{props.priceFormat(node.price)}</small>
                    <Link to={`/${node.id}`}>
                        Comprar ahora <span> --- </span>
                    </Link>
                </article>
            )}
        </section>
    </StyledProducts>
);

Products.defaultProps = {
    priceFormat: () => {},
    edges: [],
}
  
Products.propTypes = {
    priceFormat: PropTypes.func,
    edges: PropTypes.arrayOf(PropTypes.object),
}

export default Products;