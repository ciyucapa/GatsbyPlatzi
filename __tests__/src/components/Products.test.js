import React from 'react';
import renderer from 'react-test-renderer';
import Products from '../../../src/components/Products';

describe("Products", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<Products 
                priceFormat={() => 3500} 
                edges={[{
                    node: {
                        image: 'https://platzi.com/images',
                        id: 'asas',
                        price: 3500,
                        product: {
                            name: 'test'
                        } 
                    }
                }]} 
                />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    });
});