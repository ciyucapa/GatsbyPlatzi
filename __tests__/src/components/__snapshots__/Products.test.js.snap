// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products renders correctly 1`] = `
<div
  className="sc-htpNat chkFFt"
>
  <h2>
    Productos
  </h2>
  <section>
    <article>
      <img
        alt="test"
        src="https://platzi.com/images"
      />
      <p>
        test
      </p>
      <small>
        3500
      </small>
      <a
        href="/asas"
      >
        Comprar ahora 
        <span>
           --- 
        </span>
      </a>
    </article>
  </section>
</div>
`;
